### PR TITLE
fix(duration-formatter): Fixes the calculation

### LINKED ISSUE #1395: dtDuration formatter cannot handle values < 1 ms in the 'DEFAULT'-mode correctly
# Bug Report

## Expected Behavior

The dtDurationFormatter formats values smaller than 1 ms in the DEFAULT-mode correctly.

## Current Behavior

The dtDuration formatter has wrong behavior with the input- and output-units `DtTimeUnit.NANOSECOND` and `DtTimeUnit.MICROSECOND` in the 'DEFAULT'-Mode.  

See the Stackblitz example.

## Steps to Reproduce

[Stackblitz](https://stackblitz.com/edit/github-dqdbji-73uafj?file=src%2Fstyles.scss)

Gist with additional test cases which currently fail (drop-in replacement for the duration.spec.ts):  https://gist.github.com/Laubi/4f15023b9229460e2ac905327ac875b3

## Context (Environment)

I try to remove my self written component with this generic one. Although there might be further improvements needed. But this bugfix is the first step.

### Used Versions:

- **node**: 14.6.0
- **npm**: 6.14.6
- **typescript**: 3.8.3
- **angular**: 9.1
- **@angular/cdk**: 9.2.4
- **@dynatrace/barista-components**: trunk (latest release 7.5)

### PR DESCRIPTION
**Pull request by rowa-audil**
### <strong>Pull Request</strong>

The duration formatter previously used millisecond as the base unit which resulted in many workarounds due to buggy float number calculations.

Fixes #1395

#### Type of PR

Bugfix (non-breaking change which fixes an issue)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomheller** [comment]
There are still tests failing, I guess this needs some more work. I have converted it to draft.

File: libs/barista-components/formatters/src/duration/duration-formatter-constants.ts [hunk 1]
```diff
@@ -33,19 +33,19 @@ export function toDurationMode(
...
-  [DtTimeUnit.YEAR, 12 * 30.41666 * 24 * 60 * 60 * 1000],
-  [DtTimeUnit.MONTH, 30.41666 * 24 * 60 * 60 * 1000],
-  [DtTimeUnit.DAY, 24 * 60 * 60 * 1000],
-  [DtTimeUnit.HOUR, 60 * 60 * 1000],
-  [DtTimeUnit.MINUTE, 60 * 1000],
-  [DtTimeUnit.SECOND, 1000],
-  [DtTimeUnit.MILLISECOND, 1],
-  [DtTimeUnit.MICROSECOND, 1000], // Has to be handled differently because IEEE can't handle floating point numbers very well
-  [DtTimeUnit.NANOSECOND, 1], // Has to be handled differently because IEEE can't handle floating point numbers very well
+  [DtTimeUnit.YEAR, 12 * 30.41666 * 24 * 60 * 60 * 1000000000],
```
Comment (ffriedl89) [inline libs/barista-components/formatters/src/duration/duration-formatter-constants.ts:36]: These are really hard to read but there is a cool trick you can use to make it easier. 
```suggestion
  [DtTimeUnit.YEAR, 12 * 30.41666 * 24 * 60 * 60 * 1_000_000_000],
```

Comment (ffriedl89) [reply libs/barista-components/formatters/src/duration/duration-formatter-constants.ts:36]: I'd also suggest calculate them step by step. 
start with nano as 1.
Next use nano and multiply by 1000. Next use microsecond and multiply by 1000. This makes it easier to see where you are at in the chain. Store them in the map at the end.

File: libs/barista-components/formatters/src/duration/duration-formatter-constants.ts [hunk 2]
```diff
@@ -33,19 +33,19 @@ export function toDurationMode(
...
+  [DtTimeUnit.NANOSECOND, 1],
 ]);
 
 /** Default for the conversionunit when no formatmethod is passed as a number. */
 export const CONVERSIONUNITS = 3;
 
-/** Use when converting a duration to micro- or nanoseconds */
-export const MOVE_COMMA = 1000000;
+/** Converting numbers like 0.001001 won't return the intended result so the comma needs to be moved. */
+export const MOVE_COMMA = 100000000000;
```
Comment (ffriedl89) [inline libs/barista-components/formatters/src/duration/duration-formatter-constants.ts:51]: use trick here as well - Where is the comma moved? Please add it to the comment - how many decimal places are we talking and why?

File: libs/barista-components/formatters/src/duration/duration-formatter-utils/transform-result.ts [hunk 3]
```diff
@@ -38,18 +38,15 @@ export function dtTransformResult(
   const result = new Map<DtTimeUnit, string>();
   const unitsToDisplay =
     typeof formatMethod === 'number' ? formatMethod : CONVERSIONUNITS;
-  let rest = duration * CONVERSION_FACTORS_TO_MS.get(inputUnit)!;
+  let rest = dtConvertToUnit(duration, inputUnit);
+  // debugger;
```
Comment (ffriedl89) [inline libs/barista-components/formatters/src/duration/duration-formatter-utils/transform-result.ts:42]: Please cleanup

**Pull request merged**